detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ObjectLiteralToLambda.kt

```diff
@@ -0,0 +1,76 @@
+package io.gitlab.arturbosch.detekt.rules.style
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.rules.isOverride
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+import org.jetbrains.kotlin.psi.KtObjectLiteralExpression
+import org.jetbrains.kotlin.psi.KtSuperTypeEntry
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.KotlinType
+
+/**
+ * An anonymous object that does nothing other than the implementation of a single method
+ * can be used as a lambda.
+ *
+ * See https://kotlinlang.org/docs/fun-interfaces.html
+ *
+ * <noncompliant>
+ * object : Foo {
+ *     override fun bar() {
+ *     }
+ * }
+ * </noncompliant>
+ *
+ * <compliant>
+ * Foo {
+ * }
+ * </compliant>
+ *
+ * @requiresTypeResolution
+ */
+class ObjectLiteralToLambda(config: Config = Config.empty) : Rule(config) {
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Report object literals that can be changed to lambdas.",
+        Debt.FIVE_MINS
+    )
+
+    private val KotlinType.isSamInterface
```

Comment:
```suggestion
    private val KotlinType.couldBeSamInterface
```

According to its [original documentation](
https://github.com/JetBrains/kotlin/blob/master/core/descriptors/src/org/jetbrains/kotlin/descriptors/ClassDescriptor.java#L112), `isDefinitelyNotSamInterface` may return falsely false

Reply:
done